Src/IronPython.Modules/_thread.cs

```diff
@@ -23,6 +24,7 @@ public static class PythonThread {
 
         private static readonly object _stackSizeKey = new object();
         private static object _threadCountKey = new object();
+        private static ConcurrentDictionary<int, Lazy<@lock>> _sentinelTable = new ConcurrentDictionary<int, Lazy<@lock>>();
```

Comment:
Might be a good candidate for a `ThreadStatic`?
```C#
[ThreadStatic]
private static @lock _sentinel;
```

Reply:
Never knew that. Replaced with it in commit 972a7bd.